### PR TITLE
Support Increments as Float Numbers

### PR DESCRIPTION
**Pull request by Nakaner**
There are use cases where you extract more than one frame per second from a video stream, i.e. if you [film out of a 120 kph fast train](https://www.openstreetmap.org/user/Nakaner/diary/37524).

### CODE REVIEW

**Comment by Nakaner** [comment]
Are there any objections against merging this pull request?

File: python/geotag_from_gpx.py [hunk 1]
```diff
@@ -109,7 +109,13 @@ def exif_time(filename):
...
     m = pyexiv2.ImageMetadata(filename)
     m.read()
-    return m['Exif.Photo.DateTimeOriginal'].value
+    try:
+        image_time = m['Exif.Photo.DateTimeOriginal'].value
+        image_millisec = int(m['Exif.Photo.SubSecTimeOriginal'].value)
+    except ValueError, e:
+        print("Error reading EXIF at {0}: {1}".format(filename, e))
+    image_time = image_time + datetime.timedelta(milliseconds=int(image_millisec))
+    return image_time
```
Comment (ybkuang) [inline python/geotag_from_gpx.py]: Good to add error handling here! Line #117 can still fail if Exif.Photo.SubSecTimeOriginal is missing. Maybe we can set it to be default 0?

**Comment by ybkuang** [comment]
@Nakaner Sorry about the delay.

File: python/geotag_from_gpx.py (hunk 1 above)
Comment (Nakaner) [inline python/geotag_from_gpx.py]: I have added a `image_millisec = 0` and handling of KeyErrors.

File: python/geotag_from_gpx.py [hunk 2]
```diff
@@ -109,7 +109,16 @@ def exif_time(filename):
     '''
     m = pyexiv2.ImageMetadata(filename)
     m.read()
-    return m['Exif.Photo.DateTimeOriginal'].value
+    image_millisec = 0
```
Comment (ybkuang) [inline python/geotag_from_gpx.py]: @Nakaner Good find with the missing support of sub seconds. We've also tried to solve this https://github.com/mapillary/mapillary_tools/blob/master/python/lib/exif.py#L166. Would you like to help replace this part with `extract_capture_time`?

**Comment by Nakaner** [comment]
I am not going to sing this agreement for a stale pull request none of the maintainers has shown interest for 4½ years. The upstream code has basically been rewritten since then.

**Pull request closed without merging**